derive/src/lib.rs

```diff
@@ -140,6 +138,25 @@ fn crate_name_ident(name: &str) -> Result<Ident> {
         .map_err(|e| syn::Error::new(Span::call_site(), &e))
 }
 
+/// Find the name given to the `scale-info` crate in the context we run in.
+/// If scale-info is not among the dependencies then we must be deriving
+/// types for the scale-info crate itself, in which case we need to rename
+/// "self" to something, so the object paths keep working.
+fn scale_info_crate_tokens() -> Result<TokenStream2> {
```

Comment:
We could return `syn::Path` via `parse_quote!` here instead of a `TokenStream`, making it a little more strongly typed, but not necessary.

Reply:
Then we could also rename the function to `scale_info_crate_path` which is also more appealing and reusable.